### PR TITLE
chore(js): moved sprintf bower-asset into composer dependencies

### PR DESCRIPTION
**Pull request by jdalsem**
Also updated to v1.0.3

### CODE REVIEW

File: views/default/elgg.js.php [hunk 1]
```diff
@@ -13,8 +13,10 @@
 
 JS;
 
-// For backwards compatibility...
-echo elgg_view('sprintf.js');
+?>
+//<script>
+require(['sprintf']);
```
Comment (jdalsem) [inline views/default/elgg.js.php]: i think so too... it works, but i think i got lucky... would be nice to loose all custom libraries, so languages would just depend on sprintf

Comment (mrclay) [inline views/default/elgg.js.php]: Async echo would allow this #9028. Inlining this one function isn't a big deal.

**Comment by jdalsem** [comment]
Does anybody know how to fix the karma issues? @Elgg/core

**Comment by mrclay** [comment]
Problem is the simple AMD wrapper. In the Karma environment `define.amd` is defined and so we can't load the file outside of RequireJS. jQuery's wrapper is smarter. I'm still trying to figure this out.

**Comment by hypeJunction** [comment]
Try `elgg_define_js()` with an `exports` value set. That helps in some cases.

**Comment by jdalsem** [comment]
Had to do some research into karma... never worked with it before... but as the sprintf.js could not be included as a static file anymore (due to the `Mismatched anonymous define() module` error) it had to be required as a module. But because the module is not physically present as `sprintf` (due to it being declared in views.php) we had to require it using its full path... luckily that worked :)

**Pull request merged**